core/src/BibleEngine.class.ts

```diff
@@ -250,6 +249,20 @@ export class BibleEngine {
         return db.save(new BibleVersionEntity(version));
     }
 
+    async updateVersion(version: IBibleVersion) {
```

Comment:
in order to be very clear and specific what this does (compared to `saveVersion` which also updates) i'd recommend renaming it and explicitly add the `uid` parameter: `updateVersionByUid(uid: string, data: Partial<IBibleVersion>)`

this would break your `!existingVersion` part, which you can fix by making the type not partial. however i'd rather throw an error when a method can't do what it is supposed to do.

Reply:
So, the reason I require the full version to be provided is because this doesn't support partial updates (PATCH).

In other words, if some fields are left undefined or null, in Partial<IBibleVersion>, they will be overwritten by the QueryBuilder update.

Thoughts?

Reply:
if you require a full version object to be passed to the method i wonder what the use case for updating? if you try to add a version that is already there i would think that there is an error somewhere else. in other words: either you are updating a version (and then you can pass a partial object) or you are adding a version - in the latter case i think addVersion should be used and there needs to be a failing check for uid to exist at some point. one could argue that that can be added to addVersion which throws an error when attempting to add a version with existing uid.